### DISCUSSION TITLE
How to store data into a Map from json file

### DISCUSSION THREAD

**Discussion by PrithviDuggirala** [nlohmann/json#2692] [Q&A] [answered]
#### What is the issue you have?

I'm having a json file as follows:

```json
[
{
      "top__DOT__comp_ac_Multiplier_Mux01__DOT__H11TW040L_UCVCSND2X010_Z_s": [
        "sa0", 
        "sa1"
      ], 
      "top__DOT__comp_ac_Multiplier_Mux01__DOT__INDEX02_Outp_s": [
        "sa0", 
        "sa1"
      ], 
      "top__DOT__comp_ac_Multiplier_Mux01__DOT__comp_H11TW040L_UCVCSND2X010__DOT__NOT00_Outp_s": [
        "sa0"
      ], 
      "top__DOT__comp_ac_Multiplier_Mux01__DOT__comp_H11TW040L_UCVCSND2X010__DOT__NOT01_Outp_s": [
        "sa0"
      ]
    }, 
    {
      "top__DOT__comp_ac_Multiplier_Mux01__DOT__H11TW040L_UCVCSAORI1X010_Z_s": [
        "sa0", 
        "sa1"
      ], 
      "top__DOT__comp_ac_Multiplier_Mux01__DOT__INDEX00_Outp_s": [
        "sa1"
      ], 
      "top__DOT__comp_ac_Multiplier_Mux01__DOT__INDEX01_Outp_s": [
        "sa1"
      ], 
      "top__DOT__comp_ac_Multiplier_Mux01__DOT__comp_H11TW040L_UCVCSAORI1X010__DOT__BAND_Outp_s": [
        "sa0"
      ], 
      "top__DOT__comp_ac_Multiplier_Mux01__DOT__comp_H11TW040L_UCVCSAORI1X010__DOT__NOT_Outp_s": [
        "sa0"
      ]
    }
  ]
```

It is a list of dictionaries. I want to read this file and store each dictionary into a map of vectors. Later I have to iterate over the each dictionary and extract key and values(which is a list in this case). 

I have used the following code to do this:

```cpp
class faults{

    map<string, vector<string>> myMap;

};

std::ifstream myinput("fault_list.json");
auto j = json::parse(myinput);
std::vector<faults> fault = j;
```

I'm facing the following error:
error: conversion from 'nlohmann::basic_json<>' to non-scalar type 'std::vector<faults>' requested
 std::vector<faults> fault = j;
                                 ^

Operating System: Unix
Compiler: g++ (GCC) 8.2.0

I used the json.hpp file from here
https://github.com/nlohmann/json/releases/tag/v3.9.1

Kindly help me with the issue.

**Comment by nlohmann** [comment, marked as answer]
You need to define a conversion from JSON to your type `faults`. See https://github.com/nlohmann/json#arbitrary-types-conversions for guidance.

**Reply by PrithviDuggirala** [reply]
Thank you for such a quick reply. I'm pretty new to c++. I have already gone through the page you suggested and it wasn't very clear to me. Can you please, write the lines I need to include to make my code work. It'll be very helpful to me.
